grpc/balancer.go

```diff
@@ -0,0 +1,44 @@
+package grpc
+
+import (
+	"google.golang.org/grpc/naming"
+)
+
+type staticResolver struct {
+	addresses []string
+}
+
+// Resolve is called only once by grpc.RoundRobin.Start, it constructs and returns the
+// naming.Watcher which actually provides addresses to grpc.Dial
+func (sr *staticResolver) Resolve(target string) (naming.Watcher, error) {
+	sw := &staticWatcher{}
+	for _, a := range sr.addresses {
```

Comment:
I think you can slightly simplify things by having staticResolver satisfy both the `naming.Resolver` and `naming.Watcher` interfaces. Then instead of defining an additional intermediate type, you just construct the `[]*naming.Update` in `Next()`. `Resolve()` could then be empty.